app/helpers/application_helper.rb

```diff
@@ -266,7 +266,7 @@ def edit_select(object, property, options = {})
 
   def flot_pie_chart(name, title, data, options = {})
     data = data.map { |k,v| {:label=>k.to_s.humanize, :data=>v} } if data.is_a?(Hash)
-    data.map{|element| element[:label] = truncate(element[:label],:length => 16)}
+    data = Hash[data.keys.map { |name| [ truncate(name, :length => 16), data[name] ] }]
```

Comment:
any idea why the data is stored like that to begin with? doesn't it happen in other places? (facts)
/cc @amirfefer i think this is the issue we saw a few days ago

Reply:
It seems that this would only work for charts generated in config reports page, e.g. compliance breakdown widget would not work since there we send data in this expected format. I was unable to find the cause but for some reason, config report data are sent as pure hash

e.g.
```
{ 'a' => 3 }
```

while elsewhere we get this format
```
[{:label=>"Failed", :data=>0.0, :color=>"#AA4643"}]
```

Reply:
Then maybe we should fix config reports to send the data in the correct format and simplify this method?

Reply:
ofc, I'll try to find out more details first... I saw similar error elsewhere, it might be related to Rails 5 and places in Foreman where we serialize data, ActionController::Parameters is no longer inheriting from Hash but I'll need more time to investigate

Reply:
maybe the remaning failures in #4836 (or most of them) are related to that topic?